src/replication/DealReplicationWorker.ts

```diff
@@ -0,0 +1,327 @@
+        if (!useLotus && !model.isOffline) {
+          try {
+            await axios.head(downloadUrl);
+          } catch (error) {
+            this.logger.error(`This download URL ${downloadUrl} is not reachable.`);
+            continue;
+          }
+        }
+
+        let dealCmd = '';
+        if (useLotus) {
+          if (model.isOffline) {
+            const unpaddedSize = carFile.pieceSize! * 127 / 128;
+            dealCmd = `${lotusCMD} client deal --manual-piece-cid=${carFile.pieceCid} --manual-piece-size=${unpaddedSize} ` +
+              `--manual-stateless-deal --from=${model.client} --verified-deal=${model.isVerfied} ` +
+              `${carFile.dataCid} ${miner} ${priceInFil.toNumber()} ${model.duration}`;
+          } else {
+            this.logger.error(`Deal making failed. SP ${miner} only supports lotus and for lotus we only support offline deals.`);
+            return;
+          }
+        } else {
+          let propose = `deal --http-url=${downloadUrl}`;
+          if (model.isOffline) {
+            propose = `offline-deal`;
+          }
+          dealCmd = `${boostCMD} ${propose} --provider=${miner}  --commp=${carFile.pieceCid} --car-size=${carFile.carSize} ` +
+            `--piece-size=${carFile.pieceSize} --payload-cid=${carFile.dataCid} --storage-price-per-epoch=${priceInAtto} ` +
+            `--verified=${model.isVerfied} --wallet=${model.client} --duration=${model.duration}`;
+        }
+
+        this.logger.debug(dealCmd);
+        let dealCid = 'unknown';
+        let errorMsg = '';
+        let state = 'proposed';
+        let retryCount = 0;
+        do {
+          try {
+            const cmdOut = await exec(dealCmd);
+            this.logger.info(cmdOut.stdout);
+            if (useLotus) {
+              if (cmdOut.stdout.startsWith('baf')) {
+                dealCid = cmdOut.stdout;
+                dealsMadePerSP++;
+                break; // success, break from while loop
+              } else {
+                errorMsg = cmdOut.stdout + cmdOut.stderr;
+                state = 'error';
+              }
+            } else {
+              const match = boostResultUUIDMatcher.exec(cmdOut.stdout);
+              if (match != null && match.length > 1) {
+                dealCid = match[1];
+                dealsMadePerSP++;
+                break; // success, break from while loop
+              } else {
+                errorMsg = cmdOut.stdout + cmdOut.stderr;
+                state = 'error';
+              }
+            }
+          } catch (error) {
+            this.logger.error('Deal making failed', error);
+            errorMsg = '' + error;
+            state = 'error';
+          }
+          this.logger.info('Waiting 30 seconds to retry');
+          // TODO expoential back off
+          await new Promise(resolve => setTimeout(resolve, 30000));
+          retryCount++;
+        } while (retryCount < 3);
+        await Datastore.DealStateModel.create({
```

Comment:
Should we only create the entry if dealCid != 'unknown'?

Reply:
I wanted to create record even if it's error because it can record what the error message is.